File: robostream/lib/services/slave_calculator.dart
class SlaveCalculator {
  final int totalScreens;

  SlaveCalculator({required this.totalScreens});

  /// Calculates the leftmost screen number where the logo should be displayed
  /// Formula: totalScreens // 2 + 2
  int get leftmostScreen => totalScreens ~/ 2 + 2;

  /// Calculates the rightmost screen number where data and camera should be displayed
  /// Formula: totalScreens // 2 + 1
  int get rightmostScreen => totalScreens ~/ 2 + 1;

  /// Validates if the total screens number is valid
  bool get isValidScreenCount => totalScreens > 0 && totalScreens % 2 == 1;

  /// Gets all screen numbers from 1 to totalScreens
  List<int> get allScreens => List.generate(totalScreens, (index) => index + 1);

  /// Gets the middle screen number (master screen)
  int get masterScreen => (totalScreens ~/ 2) + 1;

  @override
  String toString() {
    return 'SlaveCalculator(totalScreens: $totalScreens, leftmost: $leftmostScreen, rightmost: $rightmostScreen, master: $masterScreen)';
  }
}

File: Server/FastAPI/LG/slave_calculator.py
class SlaveCalculator:
    def __init__(self, total_screens: int):
        self.total_screens = total_screens
        
    @property
    def leftmost_screen(self) -> int:
        return self.total_screens // 2 + 2
    
    @property
    def rightmost_screen(self) -> int:
        return self.total_screens // 2 + 1
    
    @property
    def master_screen(self) -> int:
        return (self.total_screens // 2) + 1
    
    @property
    def is_valid_screen_count(self) -> bool:
        return self.total_screens > 0 and self.total_screens % 2 == 1
    
    @property
    def all_screens(self) -> list:
        return list(range(1, self.total_screens + 1))
    
    def __str__(self) -> str:
        return f'SlaveCalculator(total_screens: {self.total_screens}, leftmost: {self.leftmost_screen}, rightmost: {self.rightmost_screen}, master: {self.master_screen})'
